Frontend/ui-driver/src/Screens/DriverReferralScreen/View.purs

```diff
@@ -0,0 +1,341 @@
+import Effect.Class (liftEffect)
+import Debug (spy)
+import Mobility.Prelude
+import Engineering.Helpers.BackTrack (liftFlowBT)
+
+screen :: DriverReferralScreenState -> Screen Action DriverReferralScreenState ScreenOutput
+screen initialState =
+  { initialState
+  , view
+  , name: "DriverReferralScreen"
+  , globalEvents: [
+              ( \push -> do
+                      _ <- launchAff $ flowRunner defaultGlobalState $ runExceptT $ runBackT $ do
+                        (ReferredDriversResp referredDriversResp) <- Remote.referredDriversBT (ReferredDriversReq "")
+                        liftFlowBT $ push $ ReferredDriversAPIResponseAction referredDriversResp.value
+                      pure $ pure unit
+              )
+  ]
+  , eval:
+      ( \action state -> do
+          let _ = spy "DriverReferralScreen ----- state" state
+          let _ = spy "DriverReferralScreen --------action" action
+          eval action state
+      )
+  }
+
+
+view :: forall w. (Action -> Effect Unit) -> DriverReferralScreenState -> PrestoDOM (Effect Unit) w
+view push state =
+  relativeLayout
+  [ height MATCH_PARENT
+  , width MATCH_PARENT
+  , orientation VERTICAL
+  , gravity CENTER
+  , onBackPressed push $ const BackPressed
+  , afterRender push $ const AfterRender
+  , background Color.white900
+  ][
+    linearLayout
+    [ height MATCH_PARENT
+    , width MATCH_PARENT
+    , orientation VERTICAL
+    ][  linearLayout
+        [ width MATCH_PARENT
+        , orientation VERTICAL
+        , weight 1.0
+        , height WRAP_CONTENT
+        ]
+        [ linearLayout
+          [ width MATCH_PARENT
+          , height WRAP_CONTENT
+          , orientation HORIZONTAL
+          ][ GenericHeader.view (push <<< GenericHeaderActionController) (genericHeaderConfig state)]
+          , linearLayout
+            [ height $ V 1
+            , width MATCH_PARENT
+            , background Color.greySmoke
+            ][]
+          , linearLayout
+            [ width MATCH_PARENT
+            , height WRAP_CONTENT
+            , margin $ Margin 15 20 15 15
+            , orientation VERTICAL
+            ][ driverReferralCode push state
+             , referredDriversView push state
+             , rideLeaderBoard push state
+             ]
+          , qrScreenView push state
+          ]
+          , bottomNavBarView push state]
+          , if state.props.showDriverReferralQRCode then appQRCodeView push state  else linearLayout[][]
+  ]
+
+
+driverReferralCode :: forall w . (Action -> Effect Unit) -> DriverReferralScreenState -> PrestoDOM (Effect Unit) w
+driverReferralCode push state =
+    linearLayout
+    [ height WRAP_CONTENT
+    , width MATCH_PARENT
+    , orientation VERTICAL
+    , padding $ Padding 18 15 8 15
+    , background state.data.config.primaryTextColor
+    , cornerRadius 16.0
+    ][ linearLayout
+        [ width MATCH_PARENT
+        , height WRAP_CONTENT
+        , orientation VERTICAL
+        , gravity CENTER
```

Comment:
instead of using `gravity CENTER` in every view try to use `layoutGravity "center"` in single parent view